examples/move/reviews_rating/sources/service.move

```diff
@@ -0,0 +1,301 @@
+            let reward = coin::from_balance(sub_balance, ctx);
+            let review_id = vector::borrow(&service.top_reviews, i);
+            let record = df::borrow<ID, ReviewRecord>(&service.id, *review_id);
+            transfer::public_transfer(reward, record.owner);
+            i = i + 1;
+        };
+    }
+
+    /// Adds coins to reward pool
+    public fun top_up_reward(
+        service: &mut Service,
+        coin: Coin<SUI>
+    ) {
+        balance::join(&mut service.reward_pool, coin::into_balance(coin));
+    }
+
+    /// Mints a proof of experience for a customer
+    public fun generate_proof_of_experience(
+        cap: &AdminCap,
+        service: &Service,
+        recipient: address,
+        ctx: &mut TxContext
+    ) {
+        // generate an NFT and transfer it to customer who can use it to write a review with higher score
+        assert!(cap.service_id == object::uid_to_inner(&service.id), EInvalidPermission);
+        let poe = ProofOfExperience {
+            id: object::new(ctx),
+            service_id: cap.service_id
+        };
+        transfer::transfer(poe, recipient);
+    }
+
+    /// Removes a review (only moderators can do this)
+    public fun remove_review(
+        _: &Moderator,
+        service: &mut Service,
+        review_id: ID,
+    ) {
+        assert!(service.reviews.contains(review_id), ENotExists);
+        let record = df::remove<ID, ReviewRecord>(&mut service.id, review_id);
+        service.overall_rate = service.overall_rate - (record.overall_rate as u64);
+        let (contains, i) = service.top_reviews.index_of(&review_id);
+        if (contains) {
+            service.top_reviews.remove(i);
+        };
+        let review = object_table::remove(&mut service.reviews, review_id);
+        review::delete_review(review);
+    }
+
+    /// Reorder top_reviews after a review is updated
+    fun reorder(
+        service: &mut Service,
+        review_id: ID,
+        total_score: u64
+    ) {
+        let (contains, idx) = service.top_reviews.index_of(&review_id);
+        if (!contains) {
+            update_top_reviews(service, review_id, total_score);
+        } else {
+            // remove existing review from vector and insert back
+            service.top_reviews.remove(idx);
+            let idx = find_idx(service, total_score);
+            service.top_reviews.insert(review_id, idx);
+        }
+    }
+
+    /// Upvotes a review
+    public fun upvote(service: &mut Service, review_id: ID) {
+        let review = service.reviews.borrow_mut(review_id);
+        review.upvote();
+        service.reorder(review_id, review.get_total_score());
+    }
+
+    /// Downvotes a review
+    public fun downvote(service: &mut Service, review_id: ID) {
+        let review = service.reviews.borrow_mut(review_id);
+        review.downvote();
+        service.reorder(review_id, review.get_total_score());
+    }
```

Comment:
```suggestion
    public fun upvote(service: &mut Service, review_id: ID) {
        let review = &mut service.reviews[review_id];
        review.upvote();
        service.reorder(review_id, review.get_total_score());
    }

    /// Downvotes a review
    public fun downvote(service: &mut Service, review_id: ID) {
        let review = &mut service.reviews[review_id];
        review.downvote();
        service.reorder(review_id, review.get_total_score());
    }
```

Reply:
Thank you for Move 2024 suggestions! It sure was a pleasant experience porting to Move 2024.
I've also removed downvote (and all the references to downvote) as suggested.